src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,485 @@
+            {
+                _logger.LogInformation($"Creating a new comment for the issue : '{issue.Number}'");
+                await client.Issue.Comment.Create(repoId, issue.Number, description);
+            }
+        }
+
+        /// <summary>
+        /// Builds description and creates a func to check if the existing description has to be replaced or not.
+        /// </summary>
+        /// <param name="updateHistoryError">Error info for which github issue has to be created</param>
+        /// <param name="issueRepo">Repository where issue has to be created.</param>
+        /// <returns></returns>
+        private async Task IssueDescriptionEvaluator(RepositoryBranchUpdateHistoryEntry updateHistoryError, string issueRepo)
+        {
+            var description = "";
+            Func<string, string, bool> shouldReplaceDescription;
+            switch (updateHistoryError.Method)
+            {
+                // Only UpdateAssetAsync method has subscriptionId as one of the parameters in the arguments column. Other methods do not have this info.
+                case "UpdateAssetsAsync":
+                    description = ParseUpdateAssetsMethod(updateHistoryError);
+                    if (!string.IsNullOrEmpty(description))
+                    {
+                        shouldReplaceDescription = (argument, issueBody) =>
+                        {
+                            if (!string.IsNullOrEmpty(issueBody))
+                            {
+                                var argumentParser =  ParseIssueCommentBody(issueBody);
+                                var subId = GetSubscriptionId(updateHistoryError.Arguments);
+                                if (!string.IsNullOrEmpty(argumentParser.subscriptionId) && 
+                                    string.Equals(argumentParser.method, updateHistoryError.Method) &&
+                                    string.Equals(argumentParser.subscriptionId, subId) &&
+                                    string.Equals(argumentParser.errorMessage, updateHistoryError.ErrorMessage))
+                                {
+                                    return true;
+                                }
+                            }
+                            return false;
+                        };
+                    }
+                    else
+                    {
+                        _logger.LogInformation(
+                            "The subscriptionId is not valid so skipping UpdateAssetsAsync method.");
+                        shouldReplaceDescription = (argument, issueBody) => false;
+                    }
+                    break;
+
+                case "SynchronizePullRequestAsync":
+                    _logger.LogInformation("Skipping SynchronizePullRequestAsync method. Issue creation is not necessary for this method");
+                    shouldReplaceDescription = (argument, issueBody) => false;
+                    break;
```

Comment:
We should probably "return" here, no?  Because the message seems to imply that we shouldn't be creating a issue... but this break will fall down to line 399, and then create the issue.  This isn't the only case that does this.

Reply:
I think that _then_ means that description can't ever be null, so we don't need the null check that confused me in the Create method.

Reply:
oh sure, I will add a return here and the place where subscription Id is invalid and remove the check all together